query/src/sql/planner/binder/ddl/table.rs

```diff
@@ -369,8 +372,32 @@ impl<'a> Binder {
                     .collect();
                 (DataSchemaRefExt::create(fields), vec![])
             }
-            // TODO(leiysky): Support `CREATE TABLE AS SELECT` with specified column definitions
-            _ => Err(ErrorCode::UnImplement("Unsupported CREATE TABLE statement"))?,
+            (Some(source), Some(query)) => {
+                // e.g. `CREATE TABLE t (i INT) AS SELECT * from old_t` with columns speicified
+                let (source_schema, source_coments) =
+                    self.analyze_create_table_schema(source).await?;
+                let init_bind_context = BindContext::new();
+                let (_s_expr, bind_context) = self.bind_query(&init_bind_context, query).await?;
+                let query_fields: Vec<DataField> = bind_context
+                    .columns
+                    .iter()
+                    .map(|column_binding| {
+                        DataField::new(
+                            &column_binding.column_name,
+                            *column_binding.data_type.clone(),
+                        )
+                    })
+                    .collect();
+                let  source_fields = source_schema.fields().clone();
```

Comment:
I'm not sure but does this need a reformat?

Reply:
Yes. For `Create table  with specified columns`, the fields will be the Union of the `source` and `query` part.  I didn't found any method like append in `DataSchema` :(

Reply:
Sorry， I mistake   :(